src/praisonai-agents/praisonaiagents/agents/agents.py

```
@@ -2519,6 +2519,86 @@ def _run_with_planning(self):
+        logger = get_logger(__name__)
+        
+        # Close all agents async if they support it
+        for agent in self.agents:
+            try:
+                if hasattr(agent, 'aclose') and callable(agent.aclose):
+                    await agent.aclose()
+                    logger.debug(f"Async closed resources for agent: {agent.name}")
+                elif hasattr(agent, 'close') and callable(agent.close):
+                    agent.close()
+                    logger.debug(f"Closed resources for agent: {agent.name}")
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)